getting-started/quickstart-guides/angular.md

```diff
@@ -0,0 +1,834 @@
+
+![Resized, then rotated](<../../.gitbook/assets/angular/angular-sdk-basic-chained-2.png>)
+
+Let’s flip the order of transformation and see what happens.
+
+{% code title="src/app/app.component.ts" %}
+```js
+    ...
+    transformation: Array<Transformation> = [{
+        rt: "90"
+    },{
+        height: "300", 
+        width: "200",
+    }
+    ];
+    ...
+}
+```
+{% endcode %}
+
+**Chained Transformation URL:**
+
+```markup
+<img src="https://ik.imagekit.io/demo/tr:rt-90:h-300,w-200/default-image.jpg" _ngcontent-lvp-c15="" urlendpoint="https://ik.imagekit.io/demo/" ng-reflect-url-endpoint="https://ik.imagekit.io/demo/" ng-reflect-path="default-image.jpg" ng-reflect-transformation="[object Object],[object Object">
+```
+
+![Rotated, then resized](<../../.gitbook/assets/angular/angular-sdk-basic-chained-3.png>)
+
+## **Adding overlays to images**
+
+ImageKit.io allows you to add [text](../../features/image-transformations/overlay.md#text-overlay) and [image overlay](../../features/image-transformations/overlay.md) dynamically.
+
+For example, a text overlay can be used to superimpose text on an image. Try it like so:
+
+{% code title="src/app/app.component.ts" %}
+```js
+    ...
+    transformation: Array<Transformation> = [{
+        height: "300", 
+        width: "300",
+        overlayText: 'ImageKit',
+        overlayTextFontSize: "50",
+        overlayTextColor: '0651D5',
+    }];
+    ...
+}
+```
+{% endcode %}
+
+**Rendered HTML element:**
+
+```markup
+<img src="https://ik.imagekit.io/demo/tr:h-300,w-300,ot-ImageKit,ots-50,otc-0651D5/default-image.jpg" _ngcontent-twl-c15="" urlendpoint="https://ik.imagekit.io/demo/" ng-reflect-url-endpoint="https://ik.imagekit.io/demo/" ng-reflect-path="default-image.jpg" ng-reflect-transformation="[object Object]">
+```
+
+![Text Overlay (300x300px)](<../../.gitbook/assets/angular/angular-sdk-overlay-text.png>)
+
+## **Lazy-loading images in Angular**
+
+You can lazy load images using the `loading` prop in `ik-image` component. When you use `loading="lazy"`, all images that are immediately viewable without scrolling load normally. Those that are far below the device viewport are only fetched when the user scrolls near them.
+
+The SDK uses a fixed threshold based on the effective connection type to ensure that images are loaded early enough so that they have finished loading once the user scrolls near to them.
+
+{% hint style="info" %}
+You should always set the height and width of the image element to avoid[ layout shift](https://www.youtube.com/watch?v=4-d\_SoCHeWE) when lazy-loading images.
+{% endhint %}
+
+{% code title="src/app/app.component.ts" %}
+```jsx
+    ...
+    transformation: Array<Transformation> = [{
+        height: "300"
+        width: "400"
+    }];
+    ...
+}
+```
+{% endcode %}
+
+{% code title="src/app/app.component.html" %}
+```js
+<ik-image
+    path={{path}}
+    urlEndpoint="https://ik.imagekit.io/demo/"
+    [transformation]="transformation"
+    loading="lazy"
```

Comment:
Pass height and width attribute to avoid layout shift. Our docs says the same in above paragraph, but its missing from the example. This is applicable for all examples where we are doing lazy loading.
```
You should always set the height and width of the image element to avoid[ layout shift](https://www.youtube.com/watch?v=4-d\_SoCHeWE) when lazy-loading images.
```

Reply:
The height and width are set

`{% code title="src/app/app.component.ts" %}
```jsx
    ...
    transformation: Array<Transformation> = [{
        height: "300"
        width: "400"
    }];
    ...
}
```
{% endcode %}

{% code title="src/app/app.component.html" %}
```js
<ik-image
    path={{path}}
    urlEndpoint="https://ik.imagekit.io/demo/"
    [transformation]="transformation"
    loading="lazy"
    >
  </ik-image>
```
{% endcode %}`

Reply:
height and width are set in transformation, that means the resulting image will be of particular height and width. However the height and width also needs to be set on `img` tag to avoid layout shift in case of lazy loading, as browser don't know what will be the final dimension of image until loading is complete. Please read react readme.

> You should always set the height and width of the image element to avoid [layout shift](https://www.youtube.com/watch?v=4-d_SoCHeWE) when lazy-loading images.

<img width="884" alt="image" src="https://user-images.githubusercontent.com/45416977/218930494-b15a2c80-2ded-4d72-9545-a3455af07ff5.png">

Reply:
Updated